catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -293,6 +300,36 @@ public void testRouteCreationMissingMonitoredDirectory() throws Exception {
         empty());
   }
 
+  @Test
+  public void testInterruptedWhileWaitingForInputTransformers() throws InterruptedException {
+    CdmServiceWaiter serviceWaiter = mock(CdmServiceWaiter.class);
+    doThrow(new InterruptedException()).when(serviceWaiter).waitFor(anySet());
+
+    CdmState cdmState = new CdmState();
+    ContentDirectoryMonitor monitor =
+        new ContentDirectoryMonitor(
+            camelContext,
+            mock(AttributeRegistry.class),
+            1,
+            1,
+            Runnable::run,
+            serviceWaiter,
+            cdmState);
+    monitor.systemSubjectBinder = exchange -> {};
+    monitor.setNumThreads(1);
+    monitor.setReadLockIntervalMilliseconds(1000);
+
+    submitConfigOptions(monitor, monitoredDirectoryPath, ContentDirectoryMonitor.MOVE);
+    assertThat(
+        "The content directory monitor should have no route definitions",
+        monitor.getRouteDefinitions(),
+        hasSize(0));
+    assertThat(cdmState.isRoutesStarted(), is(false));
+
+    // Clear interrupted status to prevent other tests from failing
+    Thread.interrupted();
```

Comment:
I'm not sure how to get around this.

Reply:
❗️ The problem here is that if the tests fails before you reach that statement, you will still impact the other tests. With spock tests, codice-test adds a new `@ClearInterruptions` annotation to do it automatically. In JUnit, I would put that in `@After` method or as a last resort, in a finally.